File: java8-okhttp-gson/src/main/java/com/sphereon/sdk/blockchain/proof/model/StoredSettings.java
/*
 * Blockchain Proof
 * <b>With the Blockchain Proof API it is easy to prove or disprove existence of (binary) data at a certain point in time. Behind the scenes it stores entries using the Factom (bitcoin), Multichain or Ethereum blockchain by means of our generic blockchain API.</b>    The flow is generally as follows:  1. Make sure a configuration is present  2. Register content by uploading a file, some content, or providing a Stream Location from the Storage API. When you upload content you have to tell the API whether the data has already been hashed or not. If not, or when uploading a file or stream location, the API will take care of the hashing  3. Verify content by uploading a file, some content, or providing a Stream Location from the Storage API. When you upload content you have to tell the API whether the data has already been hashed or not. If not, or when uploading a file or stream location, the API will take care of the hashing. You will get back whether the content has been registered previously or not      <b>Interactive testing: </b>A web based test console is available in the <a href=\"https://store.sphereon.com\">Sphereon API Store</a>
 *
 * OpenAPI spec version: 0.9
 * Contact: dev@sphereon.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package com.sphereon.sdk.blockchain.proof.model;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import com.sphereon.sdk.blockchain.proof.model.ChainConfiguration;
import com.sphereon.sdk.blockchain.proof.model.ChainSettings;
import com.sphereon.sdk.blockchain.proof.model.CommittedChain;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;
import java.time.OffsetDateTime;

/**
 * The stored chainSettings from the chainSettings chain
 */
@ApiModel(description = "The stored chainSettings from the chainSettings chain")
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2017-12-14T13:28:04.893+01:00")
public class StoredSettings {
  @SerializedName("settingsChain")
  private CommittedChain settingsChain = null;

  @SerializedName("chainSettings")
  private ChainSettings chainSettings = null;

  @SerializedName("context")
  private String context = null;

  @SerializedName("chainConfiguration")
  private ChainConfiguration chainConfiguration = null;

  @SerializedName("validFrom")
  private OffsetDateTime validFrom = null;

  @SerializedName("metaDataChain")
  private CommittedChain metaDataChain = null;

  @SerializedName("proofChain")
  private CommittedChain proofChain = null;

  public StoredSettings settingsChain(CommittedChain settingsChain) {
    this.settingsChain = settingsChain;
    return this;
  }

   /**
   * Get settingsChain
   * @return settingsChain
  **/
  @ApiModelProperty(required = true, value = "")
  public CommittedChain getSettingsChain() {
    return settingsChain;
  }

  public void setSettingsChain(CommittedChain settingsChain) {
    this.settingsChain = settingsChain;
  }

  public StoredSettings chainSettings(ChainSettings chainSettings) {
    this.chainSettings = chainSettings;
    return this;
  }

   /**
   * Get chainSettings
   * @return chainSettings
  **/
  @ApiModelProperty(required = true, value = "")
  public ChainSettings getChainSettings() {
    return chainSettings;
  }

  public void setChainSettings(ChainSettings chainSettings) {
    this.chainSettings = chainSettings;
  }

   /**
   * Get context
   * @return context
  **/
  @ApiModelProperty(required = true, value = "")
  public String getContext() {
    return context;
  }

  public StoredSettings chainConfiguration(ChainConfiguration chainConfiguration) {
    this.chainConfiguration = chainConfiguration;
    return this;
  }

   /**
   * Get chainConfiguration
   * @return chainConfiguration
  **/
  @ApiModelProperty(required = true, value = "")
  public ChainConfiguration getChainConfiguration() {
    return chainConfiguration;
  }

  public void setChainConfiguration(ChainConfiguration chainConfiguration) {
    this.chainConfiguration = chainConfiguration;
  }

   /**
   * Get validFrom
   * @return validFrom
  **/
  @ApiModelProperty(required = true, value = "")
  public OffsetDateTime getValidFrom() {
    return validFrom;
  }

  public StoredSettings metaDataChain(CommittedChain metaDataChain) {
    this.metaDataChain = metaDataChain;
    return this;
  }

   /**
   * Get metaDataChain
   * @return metaDataChain
  **/
  @ApiModelProperty(value = "")
  public CommittedChain getMetaDataChain() {
    return metaDataChain;
  }

  public void setMetaDataChain(CommittedChain metaDataChain) {
    this.metaDataChain = metaDataChain;
  }

  public StoredSettings proofChain(CommittedChain proofChain) {
    this.proofChain = proofChain;
    return this;
  }

   /**
   * Get proofChain
   * @return proofChain
  **/
  @ApiModelProperty(value = "")
  public CommittedChain getProofChain() {
    return proofChain;
  }

  public void setProofChain(CommittedChain proofChain) {
    this.proofChain = proofChain;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    StoredSettings storedSettings = (StoredSettings) o;
    return Objects.equals(this.settingsChain, storedSettings.settingsChain) &&
        Objects.equals(this.chainSettings, storedSettings.chainSettings) &&
        Objects.equals(this.context, storedSettings.context) &&
        Objects.equals(this.chainConfiguration, storedSettings.chainConfiguration) &&
        Objects.equals(this.validFrom, storedSettings.validFrom) &&
        Objects.equals(this.metaDataChain, storedSettings.metaDataChain) &&
        Objects.equals(this.proofChain, storedSettings.proofChain);
  }

  @Override
  public int hashCode() {
    return Objects.hash(settingsChain, chainSettings, context, chainConfiguration, validFrom, metaDataChain, proofChain);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class StoredSettings {\n");
    
    sb.append("    settingsChain: ").append(toIndentedString(settingsChain)).append("\n");
    sb.append("    chainSettings: ").append(toIndentedString(chainSettings)).append("\n");
    sb.append("    context: ").append(toIndentedString(context)).append("\n");
    sb.append("    chainConfiguration: ").append(toIndentedString(chainConfiguration)).append("\n");
    sb.append("    validFrom: ").append(toIndentedString(validFrom)).append("\n");
    sb.append("    metaDataChain: ").append(toIndentedString(metaDataChain)).append("\n");
    sb.append("    proofChain: ").append(toIndentedString(proofChain)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
  
}

File: csharp-net45/src/Sphereon.SDK.Blockchain.Proof/Model/StoredSettings.cs
/* 
 * Blockchain Proof
 *
 * <b>With the Blockchain Proof API it is easy to prove or disprove existence of (binary) data at a certain point in time. Behind the scenes it stores entries using the Factom (bitcoin), Multichain or Ethereum blockchain by means of our generic blockchain API.</b>    The flow is generally as follows:  1. Make sure a configuration is present  2. Register content by uploading a file, some content, or providing a Stream Location from the Storage API. When you upload content you have to tell the API whether the data has already been hashed or not. If not, or when uploading a file or stream location, the API will take care of the hashing  3. Verify content by uploading a file, some content, or providing a Stream Location from the Storage API. When you upload content you have to tell the API whether the data has already been hashed or not. If not, or when uploading a file or stream location, the API will take care of the hashing. You will get back whether the content has been registered previously or not      <b>Interactive testing: </b>A web based test console is available in the <a href=\"https://store.sphereon.com\">Sphereon API Store</a>
 *
 * OpenAPI spec version: 0.9
 * Contact: dev@sphereon.com
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = Sphereon.SDK.Blockchain.Proof.Client.SwaggerDateConverter;

namespace Sphereon.SDK.Blockchain.Proof.Model
{
    /// <summary>
    /// The stored chainSettings from the chainSettings chain
    /// </summary>
    [DataContract]
    public partial class StoredSettings :  IEquatable<StoredSettings>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="StoredSettings" /> class.
        /// </summary>
        [JsonConstructorAttribute]
        protected StoredSettings() { }
        /// <summary>
        /// Initializes a new instance of the <see cref="StoredSettings" /> class.
        /// </summary>
        /// <param name="SingleProofChain">SingleProofChain.</param>
        /// <param name="SettingsChain">SettingsChain (required).</param>
        /// <param name="ChainSettings">ChainSettings (required).</param>
        /// <param name="ChainConfiguration">ChainConfiguration (required).</param>
        /// <param name="MetaDataChain">MetaDataChain.</param>
        public StoredSettings(CommittedChain SingleProofChain = default(CommittedChain), CommittedChain SettingsChain = default(CommittedChain), ChainSettings ChainSettings = default(ChainSettings), ChainConfiguration ChainConfiguration = default(ChainConfiguration), CommittedChain MetaDataChain = default(CommittedChain))
        {
            // to ensure "SettingsChain" is required (not null)
            if (SettingsChain == null)
            {
                throw new InvalidDataException("SettingsChain is a required property for StoredSettings and cannot be null");
            }
            else
            {
                this.SettingsChain = SettingsChain;
            }
            // to ensure "ChainSettings" is required (not null)
            if (ChainSettings == null)
            {
                throw new InvalidDataException("ChainSettings is a required property for StoredSettings and cannot be null");
            }
            else
            {
                this.ChainSettings = ChainSettings;
            }
            // to ensure "ChainConfiguration" is required (not null)
            if (ChainConfiguration == null)
            {
                throw new InvalidDataException("ChainConfiguration is a required property for StoredSettings and cannot be null");
            }
            else
            {
                this.ChainConfiguration = ChainConfiguration;
            }
            this.SingleProofChain = SingleProofChain;
            this.MetaDataChain = MetaDataChain;
        }
        
        /// <summary>
        /// Gets or Sets SingleProofChain
        /// </summary>
        [DataMember(Name="singleProofChain", EmitDefaultValue=false)]
        public CommittedChain SingleProofChain { get; set; }

        /// <summary>
        /// Gets or Sets SettingsChain
        /// </summary>
        [DataMember(Name="settingsChain", EmitDefaultValue=false)]
        public CommittedChain SettingsChain { get; set; }

        /// <summary>
        /// Gets or Sets ChainSettings
        /// </summary>
        [DataMember(Name="chainSettings", EmitDefaultValue=false)]
        public ChainSettings ChainSettings { get; set; }

        /// <summary>
        /// Gets or Sets Context
        /// </summary>
        [DataMember(Name="context", EmitDefaultValue=false)]
        public string Context { get; private set; }

        /// <summary>
        /// Gets or Sets ChainConfiguration
        /// </summary>
        [DataMember(Name="chainConfiguration", EmitDefaultValue=false)]
        public ChainConfiguration ChainConfiguration { get; set; }

        /// <summary>
        /// Gets or Sets Id
        /// </summary>
        [DataMember(Name="id", EmitDefaultValue=false)]
        public string Id { get; private set; }

        /// <summary>
        /// Gets or Sets ValidFrom
        /// </summary>
        [DataMember(Name="validFrom", EmitDefaultValue=false)]
        public DateTime? ValidFrom { get; private set; }

        /// <summary>
        /// Gets or Sets MetaDataChain
        /// </summary>
        [DataMember(Name="metaDataChain", EmitDefaultValue=false)]
        public CommittedChain MetaDataChain { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class StoredSettings {\n");
            sb.Append("  SingleProofChain: ").Append(SingleProofChain).Append("\n");
            sb.Append("  SettingsChain: ").Append(SettingsChain).Append("\n");
            sb.Append("  ChainSettings: ").Append(ChainSettings).Append("\n");
            sb.Append("  Context: ").Append(Context).Append("\n");
            sb.Append("  ChainConfiguration: ").Append(ChainConfiguration).Append("\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  ValidFrom: ").Append(ValidFrom).Append("\n");
            sb.Append("  MetaDataChain: ").Append(MetaDataChain).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as StoredSettings);
        }

        /// <summary>
        /// Returns true if StoredSettings instances are equal
        /// </summary>
        /// <param name="other">Instance of StoredSettings to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(StoredSettings other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.SingleProofChain == other.SingleProofChain ||
                    this.SingleProofChain != null &&
                    this.SingleProofChain.Equals(other.SingleProofChain)
                ) && 
                (
                    this.SettingsChain == other.SettingsChain ||
                    this.SettingsChain != null &&
                    this.SettingsChain.Equals(other.SettingsChain)
                ) && 
                (
                    this.ChainSettings == other.ChainSettings ||
                    this.ChainSettings != null &&
                    this.ChainSettings.Equals(other.ChainSettings)
                ) && 
                (
                    this.Context == other.Context ||
                    this.Context != null &&
                    this.Context.Equals(other.Context)
                ) && 
                (
                    this.ChainConfiguration == other.ChainConfiguration ||
                    this.ChainConfiguration != null &&
                    this.ChainConfiguration.Equals(other.ChainConfiguration)
                ) && 
                (
                    this.Id == other.Id ||
                    this.Id != null &&
                    this.Id.Equals(other.Id)
                ) && 
                (
                    this.ValidFrom == other.ValidFrom ||
                    this.ValidFrom != null &&
                    this.ValidFrom.Equals(other.ValidFrom)
                ) && 
                (
                    this.MetaDataChain == other.MetaDataChain ||
                    this.MetaDataChain != null &&
                    this.MetaDataChain.Equals(other.MetaDataChain)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                if (this.SingleProofChain != null)
                    hash = hash * 59 + this.SingleProofChain.GetHashCode();
                if (this.SettingsChain != null)
                    hash = hash * 59 + this.SettingsChain.GetHashCode();
                if (this.ChainSettings != null)
                    hash = hash * 59 + this.ChainSettings.GetHashCode();
                if (this.Context != null)
                    hash = hash * 59 + this.Context.GetHashCode();
                if (this.ChainConfiguration != null)
                    hash = hash * 59 + this.ChainConfiguration.GetHashCode();
                if (this.Id != null)
                    hash = hash * 59 + this.Id.GetHashCode();
                if (this.ValidFrom != null)
                    hash = hash * 59 + this.ValidFrom.GetHashCode();
                if (this.MetaDataChain != null)
                    hash = hash * 59 + this.MetaDataChain.GetHashCode();
                return hash;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
